src/net_processing.cpp

```diff
@@ -5174,9 +5174,13 @@ void PeerManagerImpl::EvictExtraOutboundPeers(std::chrono::seconds now)
         NodeId worst_peer = -1;
         int64_t oldest_block_announcement = std::numeric_limits<int64_t>::max();
 
+        // We want to prevent recently connected to Onion peers from being disconnected here, protect them as long as
+        // there are more non_onion nodes than onion nodes so far
+        size_t onion_count = 0;
         m_connman.ForEachNode([&](CNode* pnode) {
             LockAssertion lock(::cs_main);
-
+            if (pnode->addr.IsTor()) onion_count++;
```

Comment:
nit: i'd always use prefix increment `++onion_count` without a good reason

Reply:
sorry, I haven't finish sentence and sent it too early.

Read my comment as:
nit: I'd always use prefix increment `++onion_count`; without a good reason `onion_count++` should not be used.

here is doc/developer-notes.md to support my words:
```
- **Miscellaneous**
  - `++i` is preferred over `i++`.
```

Reasoning from me: the implementation of `++i` almost always is simpler for complex classes and iterators than `i++` because no need to create a temporary object.
Though, optimizing compiler almost always provide a similar performance, in debug builds performance prefix vs postfix increment can be different in several times (due to extra allocations, etc).
There's a link to support my words: https://pvs-studio.com/en/blog/posts/cpp/0093/

In this particular case there's a just integer (pod type), so, that's not critical, but that's good to maintain a habit always use a prefix increment/decrement and use it all over codebase.

Secondly, I'd always use prefix increment to reduce mental load. Every time when you see a postfix increment you may start to think - why here? So, that's just easier to read code.